Sources/PointFree/Database.swift

```diff
@@ -201,20 +226,74 @@ private func fetchSubscriptionTeammates(ownerId: Database.User.Id) -> EitherIO<E
   )
 }
 
-private func updateUser(withId id: Database.User.Id, name: String, email: EmailAddress) -> EitherIO<Error, Prelude.Unit> {
+private func updateUser(
+  withId userId: Database.User.Id,
+  name: String?,
+  email: EmailAddress?,
+  emailSettings: [Database.EmailSetting.Newsletter]?
+  ) -> EitherIO<Error, Prelude.Unit> {
+
   return execute(
     """
     UPDATE "users"
-    SET "name" = $1, "email" = $2
+    SET "name" = COALESCE($1, "name"),
+        "email" = COALESCE($2, "email")
     WHERE "id" = $3
     """,
     [
       name,
-      email.unwrap,
-      id.unwrap.uuidString,
+      email?.unwrap,
+      userId.unwrap.uuidString,
     ]
     )
+    .flatMap(const(updateEmailSettings(settings: emailSettings, forUserId: userId)))
+}
+
+private func registerUser(withGitHubEnvelope envelope: GitHub.UserEnvelope) -> EitherIO<Error, Database.User?> {
+  return upsertUser(withGitHubEnvelope: envelope)
+    .flatMap { optionalUser in 
+      guard let user = optionalUser else { return pure(optionalUser) }
+
+      return updateEmailSettings(settings: Database.EmailSetting.Newsletter.allNewsletters, forUserId: user.id)
+        .map(const(optionalUser))
+  }
+}
+
+private func updateEmailSettings(
+  settings: [Database.EmailSetting.Newsletter]?,
+  forUserId userId: Database.User.Id
+  )
+  -> EitherIO<Error, Prelude.Unit> {
+
+    guard let settings = settings else { return pure(unit) }
+
+    let deleteEmailSettings = execute(
```

Comment:
updating email settings just delete all existing settings and then re-inserts :/

for a moment I tried to be smarter by deleting with a kinda of `newsletter NOT IN $1` deal, but didnt know if the postgres lib would support and so abandoned for now